sources/packages/backend/apps/api/src/testHelpers/testing-modules/testing-modules-helper.ts

```diff
@@ -35,3 +37,24 @@ export async function createTestingAppModule(): Promise<CreateTestingModuleResul
     dataSource,
   };
 }
+
+/**
+ * API root module only with config module and config controller.
+ ** This module allows to mock any environment variable including
+ ** keycloak environment variables.
+ ** This module is exclusively for Config e2e tests.
+ * @returns test config module as root application module.
+ */
+export async function createTestingConfigModule(): Promise<CreateTestingModuleResult> {
```

Comment:
I am not following why we actually need a new one for the configs. 
My first point would be, if this is going to be used only for the configs and it is used only once, it does not need to be created at this level.
My second point would be if the env variables are mocked after the `createTestingAppModule` is called, does it not work to test the configs?

Reply:
As discussed, using the first approach to have the createTestingConfigModule in the scope of spec file.